### ISSUE TITLE
Collaborating with the hashi_vault Ansible collection ?

### ISSUE THREAD

**Issue by dmsimard** [ansible-community/ansible-vault#182]
Hi @briantist and @ansible-community/hashicorp-tools o/

We recently bootstrapped a new community Ansible collection for hashicorp vault here: https://github.com/ansible-collections/community.hashi_vault

CI is in place with the help of GitHub actions for testing new PRs and release to galaxy can also be handled automatically whenever tagging a new release.

Would it make sense to include this role in the collection so users could get both the role and the plugin from the same location ?
The Ansible community team can help get things started if you'd like to collaborate.

**Comment by bbaassssiiee** [comment]
Oh that is great news! Thanks to all who helped with this!! And it makes sense to add this. Could you do something similar for the Consul content?

**Comment by felixfontein** [comment]
The collection currently only contains the `hashi_vault` plugin (moved there from community.general).

Also note that the collection requires at least Ansible 2.9.10, so you'd (officially) drop support of Ansible 2.8. (Ansible 2.8 and 2.9 ignore the minimum version field though, so in practice it can still be used, assuming no 2.9.10+ features are used.)

**Comment by bbaassssiiee** [comment]
Hmm, I rather support Ansible 2.8.

**Comment by dmsimard** [comment]
> Also note that the collection requires at least Ansible 2.9.10, so you'd (officially) drop support of Ansible 2.8. (Ansible 2.8 and 2.9 ignore the minimum version field though, so in practice it can still be used, assuming no 2.9.10+ features are used.)

@felixfontein: thanks for pointing that out, you're right.

> Hmm, I rather support Ansible 2.8.

@bbaassssiiee Ansible 2.8 is currently scheduled for EOL [January 2021](https://access.redhat.com/support/policy/updates/ansible-engine):

![Screenshot from 2020-11-30 15-59-22](https://user-images.githubusercontent.com/1291204/100664032-1a1f5680-3325-11eb-9342-c2230aa3ddaf.png)

This also roughly coincides with the tentative schedule to release [Ansible 3.0 in February](https://hackmd.io/y7BBcweNR3aRVLuMbKkDxw?view#Proposed-Schedule-for-Ansible-30):

![Screenshot from 2020-11-30 17-07-34](https://user-images.githubusercontent.com/1291204/100671488-92d6e080-332e-11eb-8203-8315dd370e8c.png)

I believe that leaves us plenty of time (even considering the upcoming holidays) to get the role in the collection without necessarily dropping anything here just yet.

**Comment by dmsimard** [comment]
> Oh that is great news! Thanks to all who helped with this!! And it makes sense to add this. Could you do something similar for the Consul content?

I'm assuming you are talking about https://github.com/ansible-community/ansible-consul ?

Looking at the community.general collection, there's a few consul modules and plugins like:
- https://github.com/ansible-collections/community.general/tree/main/plugins/modules/clustering/consul
- https://github.com/ansible-collections/community.general/blob/main/plugins/lookup/consul_kv.py

With that in mind, it could make sense to regroup the role along with the different plugins from community.general into a collection and yes, we can help bootstrapping.

**Comment by bbaassssiiee** [comment]
So if 2.8 has an EOL date in the near future, and 2.10 introduces breaking changes, will 2.9 be an LTS version given the billions of lines of Yaml out there?

**Comment by felixfontein** [comment]
Why do you say that 2.10 has a breaking change? It has as much (or little) breaking changes as any other Ansible release.

**Comment by dmsimard** [comment]
> So if 2.8 has an EOL date in the near future, and 2.10 introduces breaking changes, will 2.9 be an LTS version given the billions of lines of Yaml out there?

@bbaassssiiee no official date is written for 2.9 that I know of but it is indeed planned to be supported longer (>= 2023).
2.10 is not so much about breaking changes -- it should "just work" in most cases -- it's more about the underlying changes in how Ansible is maintained and packaged.

2.9 is the release before these changes took place.

**Comment by bbaassssiiee** [comment]
I test many repo's with tox and they are all broken by 2.10. Example tox.ini https://github.com/dockpack/base_boost/blob/master/tox.ini

**Comment by felixfontein** [comment]
@bbaassssiiee what exactly fails with Ansible 2.10 that does work with 2.9?

**Comment by bbaassssiiee** [comment]
Dunno, too many parts are moving lately, integration is getting harder by developer entropy. So 2.10 might not be the cause.

**Comment by lanefu** [comment]
> So if 2.8 has an EOL date in the near future, and 2.10 introduces breaking changes, will 2.9 be an LTS version given the billions of lines of Yaml out there?

Based on feedback I got from an insider, 2.9 will effectively be an LTS.

Most breaking changes are going to be around add-on python based modules/plugins that link to ansible code... an example is their ec2.py inventory script doesn't work.       Also with 2.10 there's a few means of distribution, but more raw packaging like pip or git  require you to follow the new module storage model and download the module packages seperately (its easy with galaxy)

The new module model and naming conventions used in 2.10+ ARE supported in 2.9, just less enforced.

TLDR; if code is refactored for 2.10, it should work with 2.9 (which will be an LTS), but not 2.8

**Comment by bbaassssiiee** [comment]
These changes could shout for a 3 version.

**Comment by felixfontein** [comment]
> Most breaking changes are going to be around add-on python based modules/plugins that link to ansible code... an example is their ec2.py inventory script doesn't work. Also with 2.10 there's a few means of distribution, but more raw packaging like pip or git require you to follow the new module storage model and download the module packages seperately (its easy with galaxy)

I don't think there are any breaking changes for modules or plugins (except removals of deprecated things, and fixes for bugs which some code might rely on). Inventory scripts or scripts that run outside of Ansible are another story - if they rely on internals that have been moved out of ansible-base, they will stop working if not adjusted correctly.

**Comment by jorhett** [comment]
Hate to restart an idle topic, but would love to see these integrated as this role would make an excellent addition to the collection.

**Comment by bbaassssiiee** [comment]
Makes sense nowadays.
